components/viz/service/display/starboard/overlay_strategy_underlay_starboard.cc

```diff
@@ -0,0 +1,199 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "components/viz/service/display/starboard/overlay_strategy_underlay_starboard.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/containers/adapters.h"
+#include "base/logging.h"
+#include "components/viz/common/quads/draw_quad.h"
+#include "components/viz/common/quads/solid_color_draw_quad.h"
+#include "components/viz/common/quads/video_hole_draw_quad.h"
+#include "components/viz/service/display/overlay_candidate_factory.h"
+#include "ui/gfx/geometry/rect_conversions.h"
+
+namespace viz {
+
+OverlayStrategyUnderlayStarboard::OverlayStrategyUnderlayStarboard(
+    OverlayProcessorUsingStrategy* capability_checker)
+    : OverlayStrategyUnderlay(capability_checker) {
+  // TODO(borongchen): remove debug log
+  LOG(ERROR) << "Cobalt uses OverlayStrategyUnderlayStarboard";
```

Comment:
nit: don't forget to remove the debug log.

Reply:
Done.